src/main/java/com/vmware/vipclient/i18n/messages/service/LocaleService.java

```diff
@@ -31,119 +34,166 @@ public LocaleService(LocaleDTO dto) {
     }
 
     public Map<String, String> getRegions(String locale){
```

Comment:
Here supported means supported by CLDR rather than product, right?
This will need the CLDR data to support this.

At last, we should make sure the locale of translation matches the locale of pattern data.

Reply:
Maybe this can't be done in the next PR/in a short time.

Because to judge if the requested locale is supported or not, we need get the supported locale list first like L3, but currently there is no API for getting supported locale list of L2 in Singleton service.